gemrb/core/Strings/StringConversion.cpp

```diff
@@ -26,77 +26,65 @@
 
 namespace GemRB {
 
+static bool IsBigEndian() {
+	const static uint16_t endiantest = 1;
+	const static bool isBigEndian = ((char *)&endiantest)[1] == 1;
+
+	return isBigEndian;
+}
+
 static String StringFromEncodedData(const ieByte* string, const EncodingStruct& encoded)
 {
-	if (!string) return L"";
-
-	bool convert = encoded.widechar || encoded.multibyte;
-	// assert that its something we know how to handle
-	// TODO: add support for other encodings?
-	assert(!convert || (encoded.widechar || encoded.encoding == "UTF-8"));
-
-	size_t len = strlen((const char*) string);
-	String dbString;
-	dbString.reserve(len);
-	size_t dbLen = 0;
-	for(size_t i=0; i<len; ++i) {
-		ieWord currentChr = string[i];
-		// we are assuming that every multibyte encoding uses single bytes for chars 32 - 127
-		if(convert && (i+1 < len) && (currentChr >= 128 || currentChr < 32) && (currentChr != '\n')) {
-			// this is a double byte char, or a multibyte sequence
-			ieWord ch = 0;
-			if (encoded.encoding == "UTF-8") {
-				size_t nb = 0;
-				if (currentChr >= 0xC0 && currentChr <= 0xDF) {
-					/* c0-df are first byte of two-byte sequences (5+6=11 bits) */
-					/* c0-c1 are noncanonical */
-					nb = 2;
-				} else if (currentChr >= 0xE0 && currentChr <= 0XEF) {
-					/* e0-ef are first byte of three-byte (4+6+6=16 bits) */
-					/* e0 80-9f are noncanonical */
-					nb = 3;
-				} else if (currentChr >= 0xF0 && currentChr <= 0XF7) {
-					/* f0-f7 are first byte of four-byte (3+6+6+6=21 bits) */
-					/* f0 80-8f are noncanonical */
-					nb = 4;
-				} else if (currentChr >= 0xF8 && currentChr <= 0XFB) {
-					/* f8-fb are first byte of five-byte (2+6+6+6+6=26 bits) */
-					/* f8 80-87 are noncanonical */
-					nb = 5;
-				} else if (currentChr >= 0xFC && currentChr <= 0XFD) {
-					/* fc-fd are first byte of six-byte (1+6+6+6+6+6=31 bits) */
-					/* fc 80-83 are noncanonical */
-					nb = 6;
-				} else {
-					Log(WARNING, "String", "Invalid UTF-8 character: {:#x}", currentChr);
-					continue;
-				}
-
-				ch = currentChr & ((1 << (7 - nb)) - 1);
-				while (--nb) {
-					ch <<= 6;
-					ch |= string[++i] & 0x3f;
-				}
-			} else {
-				ch = (string[++i] << 8) + currentChr;
-			}
-			dbString.push_back(ch);
-		} else {
-			dbString.push_back(currentChr);
-		}
-		++dbLen;
+	if (!string) return u"";
+
+	char* in = reinterpret_cast<char*>(const_cast<ieByte*>(string));
```

Comment:
if we change the incoming data, then we ought to change the signature. Is this just because of a bad `iconv` decision to not use `const`? If thats the case then this is fine.

Reply:
Hm, looks like this can be dropped indeed, probably just left over of trying to understand what iconv takes and gives.

Reply:
Interesting, I was looking at the docs that say `const char**`: https://www.gnu.org/savannah-checkouts/gnu/libiconv/documentation/libiconv-1.17/iconv.3.html

But my lib complains on both Windows and Linux:
```
C:/MSYS2/mingw64/include/iconv.h:82:15: note: candidate function not viable: no known conversion
      from 'const char **' to 'char **' for 2nd argument
extern size_t iconv (iconv_t cd,  char* * inbuf, size_t *inbytesleft, char* * outbuf, size_...
```

Reply:
Great, some context and history: https://stackoverflow.com/questions/45938990/why-does-the-iconv-function-need-a-non-const-inbuffer